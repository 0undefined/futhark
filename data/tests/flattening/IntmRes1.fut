// --
// input {
//   [ [1,2,3], [4,5,6]
//   , [6,7,8], [9,10,11]
//   ]
//   [1,2,3,4]
//   5
// }
// output {
//   [[7, 8, 9],
//    [16, 17, 18],
//    [24, 25, 26],
//    [33, 34, 35]]
// }
fun [int] addToRow ([int] xs, int y) =
  map(fn int (int x) => x+y, xs)

fun [[int]] main ([[int]] xss, [int] cs, int y) =
  map (fn [int] ([int] xs, int c) =>
         let y' = y * c + c in
         let zs = addToRow(xs,y') in
         zs
      , zip (xss,cs))
