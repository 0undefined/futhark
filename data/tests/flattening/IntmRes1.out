[[7, 8, 9],
 [16, 17, 18],
 [24, 25, 26],
 [33, 34, 35]]
